app/public/bowling_logic.js

```diff
@@ -0,0 +1,41 @@
+var BowlingGame = function () {
+  this.runningTotal = 0
+  this.frameCount = 0
+  this.isEnded = false
+};
+
+BowlingGame.prototype.startFrame = function() {
+  firstRoll = this.roll();
+  if (firstRoll < 10)
+    return ('You knocked down ' + firstRoll + ' pins. You can roll again.')
+  else {
+    this.runningTotal += 10
+    this.frameCount++
+    return 'STRIKE!'
+  };
+};
+
+BowlingGame.prototype.roll = function() {
+  var pinsHit = Math.floor(Math.random()*11);
+  return pinsHit
+};
+
+
+BowlingGame.prototype.rollAgain = function(firstRoll) {
```

Comment:
This one is handling too much. Split into separate method for second roll, handling scores etc.